packaging/specifiers.py

```diff
@@ -516,12 +516,12 @@ def _compare_not_equal(self, prospective, spec):
     @_require_version_compare
     def _compare_less_than_equal(self, prospective, spec):
         # type: (ParsedVersion, str) -> bool
-        return prospective <= Version(spec)
+        return Version(prospective.public) <= Version(spec)
```

Comment:
Hmm... I think we'd want to bring in the logic from line 506 into these functions.

Reply:
> Local version identifiers are NOT permitted in this version specifier.

You addressed this, however we should probably add a comment about this in the code. :)